### ISSUE TITLE
Shapely v1.8 deprecation warnings in advance of Shapely 2.0 release

### ISSUE THREAD

**Issue by Asteliks** [gboeing/osmnx#764]
**Is your feature proposal related to a problem?**

Iteration over multi-part geometries is deprecated and will be removed in Shapely 2.0. Use the `geoms` property to access the constituent parts of a multi-part geometry.

**Describe the solution you'd like to propose**

Updating osmnx to use the `geoms` property

**Comment by jorisvandenbossche** [comment]
Quickly checking those listed occurences, they should be straightforward to fix I think. For example, the case of 

```python
    elif merged_outer_linestrings.geom_type == "MultiLineString":
        for merged_outer_linestring in list(merged_outer_linestrings):
            ...
```

can be changed to 

```python
    elif merged_outer_linestrings.geom_type == "MultiLineString":
        for merged_outer_linestring in merged_outer_linestrings.geoms:
            ...
```

(so adding a `.geoms` to explicitly get the parts of a MultiLineString)

See https://shapely.readthedocs.io/en/stable/migration.html#multi-part-geometries-will-no-longer-be-sequences-length-iterable-indexable for more details.

> We probably should also constrain the requirements file to keep Shapely >=1.8 and <2.0 at this point

The `>=1.8` should not necessarily be needed for now, I think. Normally all fixes you would need to do for getting it work on Shapely 1.8 without warnings should still be correct code for Shapely 1.7 as well.

**Comment by gboeing** [comment]
Thanks @jorisvandenbossche!

**Comment by gboeing** [comment]
Proposed fix in #773

**Comment by gboeing** [comment]
Resolved by #773

**Comment by jihongyu-111** [comment]
> 快速检查那些列出的事件，我认为它们应该可以直接修复。例如，的情况
> 
> ```python
>     elif merged_outer_linestrings.geom_type == "MultiLineString":
>         for merged_outer_linestring in list(merged_outer_linestrings):
>             ...
> ```
> 
> 可以更改为
> 
> ```python
>     elif merged_outer_linestrings.geom_type == "MultiLineString":
>         for merged_outer_linestring in merged_outer_linestrings.geoms:
>             ...
> ```
> 
> （因此，添加 a 以显式获取多行字符串的各个部分）`.geoms`
> 
> 有关更多详细信息[，请参阅 https://shapely.readthedocs.io/en/stable/migration.html#multi-part-geometries-will-no-longer-be-sequences-length-iterable-indexable](https://shapely.readthedocs.io/en/stable/migration.html#multi-part-geometries-will-no-longer-be-sequences-length-iterable-indexable)。
> 
> > 我们可能还应该约束需求文件，以保持 Shapely >=1.8 和 <2.0 此时
> 
> 我认为，现在不一定需要。通常，为了使其在 Shapely 1.8 上正常工作而需要执行的所有修复程序（没有警告）仍应是 Shapely 1.7 的正确代码。`>=1.8`

Hello, may I ask where this code was modified？ I did not find the source file

**Comment by gboeing** [comment]
In #773

**Comment by serrayos** [comment]
What is the fix in #773 ? I do not see any solution
